### DISCUSSION TITLE
Collapsing sidebars

### DISCUSSION THREAD

**Discussion by frankvp11** [zauberzeug/nicegui#1202] [Q&A] [answered]
### Question

In many modern applications you can easily collapse a sidebar (left/right drawer) by clicking on the page outside of the drawer itself. Is there a way we can do this with nicegui? Like where a user clicks somewhere "else" on the page that isn't said drawer and it will close said drawer? 
Sorry for the vagueness in this question, it's difficult to explain as there is no relevant code (besides ui.left_drawer).  
What I'm trying to get at is, it would be very similar to the following code, except you wouldn't need to click the button, you would just click anywhere that's not the header/left_drawer. 
```py
#!/usr/bin/env python3
from nicegui import ui
with ui.header().classes(replace='row items-center') as header:
    ui.button(on_click=lambda: left_drawer.toggle(), icon='menu').props('flat color=white')
with ui.left_drawer().classes('bg-blue-100') as left_drawer:
    ui.label('Side menu')
ui.run()
```
Edit: I figured out a good example of this functionality. When you click on the top left corner in github where the three bars are at it opens a sidebar. When you click anywhere else on the screen it closes said sidebar. This is the functionality that I am trying to reproduce.

**Comment by falkoschindler** [comment, marked as answer]
Hi @frankvp11,

Since `ui.left_drawer` is based on Quasar's [QDrawer](https://quasar.dev/layout/drawer), we can look there for some prop that fits our needs. And indeed: "behavior=mobile" is what you're looking for. Normally this prop is set to "default", which switches between mobile and desktop behavior depending on the screen size. But you can enforce either mode:

```py
with ui.header().classes(replace='row items-center') as header:
    ui.button(on_click=lambda: left_drawer.toggle(), icon='menu').props('flat color=white')
with ui.left_drawer().classes('bg-blue-100').props('behavior=mobile') as left_drawer:
    ui.label('Side menu')
```

**Reply by frankvp11** [reply]
Ah, perfect. I didn't realize that this functionality existed already. I had already read up on that documentation just didn't realize that's what it meant. Thanks a lot for this.
